pyfr/plugins/integrate.py

```diff
@@ -53,21 +53,33 @@ def __init__(self, intg, cfgsect, suffix=None):
             self.outf = init_csv(self.cfg, cfgsect, ','.join(header))
 
         # Prepare the per element-type info list
-        self.eleinfo = []
+        self.eleinfo = eleinfo = []
         for (ename, eles), (eset, emask) in zip(system.ele_map.items(), rinfo):
-            # Locations of each solution point
-            ploc = eles.ploc_at_np('upts')[..., eset]
-            ploc = ploc.swapaxes(0, 1)
+            # Obtain quadrature info
+            rname = self.cfg.get(f'solver-elements-{ename}', 'soln-pts')
+            if self.cfg.hasopt(cfgsect, 'quad-deg'):
+                if self.cfg.hasopt(cfgsect, f'quad-pts-{ename}'):
```

Comment:
and

Reply:
Or alternatively you can just try and

Reply:
Or alternatively you can put the entire fragment in a try/except block, which might be cleaner.

Reply:
Here there are two main *branches*

1) If `quad-deg` is given but `quad-pts-{ename}` is not found in the input file: then use the quadrature rule name of the solution points

2) If both `quad-deg` and `quad-pts-{ename}` are provided: then use the quadrature rule name of `quad-pts-{ename}`

That's why I did not merge the two ifs. If I use an `and` condition then it means that option 1) can not work, is this OK? I'll rewrite the ifs in a `try/except` block once we've figured out if we want to be compatible with options 1) and 2).